### DISCUSSION TITLE
Can't sort alphabetically some columns in Pocketbase

### DISCUSSION THREAD

**Discussion by danilomsilva** [pocketbase/pocketbase#4466] [Q&A] [answered]
Hi there,
I am facing some issues while trying to sort alphabetically the content of the column "boletim" as you can see in the attached screenshot. The arrow icon is shown next to "boletim" indicating the content is sorted but the result of it looks not correct.

Is there anything I could try to fix this from my side or is this a pocketbase issue? 
Also, why some other columns I can sort correclty?

![image](https://github.com/pocketbase/pocketbase/assets/68205719/0ac965cd-afd3-4799-b994-c99095e80a3f)

Thank you

**Comment by ganigeorgiev** [comment]
_Other similar discussions https://github.com/pocketbase/pocketbase/discussions/2971, https://github.com/pocketbase/pocketbase/issues/4143, https://github.com/pocketbase/pocketbase/discussions/2940._

There is no support at the moment for natural sort, if that is what you are asking.
The PocketBase `sort` in the Admin UI or via the Web API translates directly to a standard SQLite `ORDER BY X ASC/DESC` statement (you can inspect the generated query with `--dev`).

As a workaround you can try to create a helper View collection with a default `ORDER BY func` as part of its statement but what `func` is it will depend on your `boletim` column values (see [Built-in scalar SQL functions](https://www.sqlite.org/lang_corefunc.html)). For example:

```sql
SELECT id, boletim, ...
FROM example
ORDER BY length(boletim)
```

**Reply by danilomsilva** [reply]
Hello again, let me provide this example:

```
export async function getManutencoes(
  userToken: User['token'],
  sortingBy: string | null,
  filter?: string,
  page?: string,
  perPage?: string
): Promise<ManutencaoResponse> {
  let url = `${process.env.BASE_API_URL}/collections/manutencao/records`;

  const queryParams = new URLSearchParams();
  if (sortingBy) queryParams.set('sort', sortingBy);
  //Auto expand record relations. Ex.: ?expand=relField1,relField2.subRelField - From Pocketbase Docs
  queryParams.set('expand', 'encarregado,equipamento');
  queryParams.set('page', page ?? '1');
  queryParams.set('perPage', perPage ?? '30');
  queryParams.set('filter', filter ?? '');

  if (queryParams.toString()) {
    url += `?${queryParams.toString()}`;
  }

    const response = await fetch(url, {
      method: 'GET',
      headers: {
        'Content-Type': 'application/json',
        Authorization: `Bearer ${userToken}`,
      },
    });
    const data = await response.json();
    return data;
}
```

if providing +boletim/-boletim it will apply the sort ASC/DESC but it still returns the data in the wrong order. Instead of BOL-1, BOL-2, BOL-10, BOL-20, it returns BOL-1, BOL-10, BOL-2, BOL-20...

I didn't quite understand how to apply what you suggested previously. Could you please share some example on how to create a helper view? Bear in mind that I am using pocket base rest api not the sdk.

Thanks!

**Reply by ganigeorgiev** [reply, marked as answer]
> if providing +boletim/-boletim it will apply the sort ASC/DESC but it still returns the data in the wrong order

It is not wrong, there is just no support out of the box for [natural sort](https://en.wikipedia.org/wiki/Natural_sort_order) and SQLite applies a simple alphanumerical sorting.

If all your fields are either `-` or prefixed with `BOL-` one possible approach for the view query could be to trim the prefix and cast the column to int.
A [View collection](https://pocketbase.io/docs/collections/#view-collection) could be created from the _Admin UI > + New collection > Type: View_ and one example query could be:

```sql
-- ex. naturalSortedManutencao
SELECT
  id,
  boletim,
  ... list the other fields of your manutencao collection ...
FROM manutencao
ORDER BY CAST(replace(boletim, "BOL-", "") as INT)
```

Then to fetch the sorted entries you'll have to send request to the View collection endpoint:
```js
// note1: don't specify an explicit sort parameter as it will overwrite the default one from the query
// note2: View collections have their own API rules
GET /api/collections/naturalSortedManutencao/records
```

**Reply by danilomsilva** [reply]
That's all I needed and it works as expected now. 
Thanks a million for the help !
